### PR TITLE
Feature/milestone4

### PR DESCRIPTION
**Pull request by gonjavi**
This pull request: 

1. Enhance the Redux store
The Redux store should keep information about the current books filter. Name it filter. It can be a simple string with the name of the category. Remember to add the option All which will just remove the filter from the table.
Create the CHANGE_FILTER action in actions/index.js
Create a filter.js reducer which modifies filter in the Redux store. It should default to All.
2. Implement the CategoryFilter component
The component should render a select box with filter categories.
Remember to merge the book categories with the All option in the select box. All is not a valid book category and shouldn't be saved in the Redux store, although it should be a filter option.
3. Keep the filter value in the Redux store
Render the CategoryFilter component in the BooksList component.
Implement the handleFilterChange method which modifies the filter state in the Redux store. Pass it down as a prop to CategoryFilter.
Implement a conditional method which decides which books should be shown in the table depending on the filter value from Redux store.

### CODE REVIEW

File: src/containers/CategoryFilter.js [hunk 1]
```diff
@@ -0,0 +1,57 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import Form from 'react-bootstrap/Form';
+import { changeFilter } from '../actions/index';
+
```
Comment (shloch) [inline src/containers/CategoryFilter.js]: this is a presentational component, please relocate it inside `src/components/` folder
[see [requirements](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/react_reducx/bookstore/milestone4/presentationalCategoryFilter.png)]

File: src/containers/CategoryFilter.js [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+              <option>Action</option>
+              <option>Biography</option>
+              <option>History</option>
+              <option>Horror</option>
+              <option>Kids</option>
+              <option>Learning</option>
+              <option>Sci-Fi</option>
+            </Form.Control>
+          </Form.Group>
+        </Form>
```
Comment (shloch) [inline src/containers/CategoryFilter.js]: first time am seeing this syntax `Form.Control` haha

File: src/containers/BooksList.js [hunk 3]
```diff
@@ -18,30 +19,49 @@ class BooksList extends React.Component {
...
-            <th>Title</th>
-            <th>Category</th>
-            <th>Delete</th>
-          </tr>
-        </thead>
-        {BooksList}
-      </Table>
+      <div>
+        <CategoryFilter />
+        <Table striped bordered hover>
```
Comment (shloch) [inline src/containers/BooksList.js]: `handleFilterChange` should be passed to `<CategoryFilter/>` as `prop`
[see [requirements](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/react_reducx/bookstore/milestone4/HandelChangeFilter.png)]

**Review by shloch — CHANGES_REQUESTED** [review]
## Hello @gonjavi , 

We're on good track....minor corrections below please before final thumbs up
(You working alone on this project ?? )

**Comment by gonjavi** [comment]
Yes, I'm working alone for the whole project

File: src/containers/CategoryFilter.js (hunk 1 above)
Comment (gonjavi) [reply src/containers/CategoryFilter.js]: Perfect

File: src/containers/BooksList.js (hunk 3 above)
Comment (gonjavi) [reply src/containers/BooksList.js]: Perfect.

**Review by Oghenebrume50 — APPROVED** [review]
### APPROVED

Hi kudos for making changes as requested by the previous reviewer, also I must commend how detailed your PR is, **please merge and move on** :slightly_smiling_face:  **This is approved**

**RAPHAEL NORIODE** :mask:

**Pull request merged**